lib/Frontend/nvqpp/ConvertStmt.cpp

```diff
@@ -317,19 +317,47 @@ bool QuakeBridgeVisitor::VisitReturnStmt(clang::ReturnStmt *stmt) {
       if (failed(irBuilder.loadIntrinsic(module, "__nvqpp_vectorCopyCtor")))
         module.emitError("failed to load intrinsic");
       auto eleTy = vecTy.getElementType();
-      auto ptrTy = cudaq::cc::PointerType::get(builder.getI8Type());
-      Value resBuff = builder.create<cc::StdvecDataOp>(loc, ptrTy, result);
-      std::size_t byteWidth = (eleTy.getIntOrFloatBitWidth() + 7) / 8;
-      Value dynSize =
-          builder.create<cc::StdvecSizeOp>(loc, builder.getI64Type(), result);
-      auto eleSize = builder.create<arith::ConstantIntOp>(loc, byteWidth, 64);
-      Value heapCopy =
-          builder
-              .create<func::CallOp>(loc, ptrTy, "__nvqpp_vectorCopyCtor",
-                                    ValueRange{resBuff, dynSize, eleSize})
-              .getResult(0);
-      result = builder.create<cc::StdvecInitOp>(loc, resTy,
+      auto createVectorInit = [&](Value eleSize) {
+        auto ptrTy = cudaq::cc::PointerType::get(builder.getI8Type());
+        Value resBuff = builder.create<cc::StdvecDataOp>(loc, ptrTy, result);
+        Value dynSize =
+            builder.create<cc::StdvecSizeOp>(loc, builder.getI64Type(), result);
+        Value heapCopy =
+            builder
+                .create<func::CallOp>(loc, ptrTy, "__nvqpp_vectorCopyCtor",
+                                      ValueRange{resBuff, dynSize, eleSize})
+                .getResult(0);
+        return builder.create<cc::StdvecInitOp>(loc, resTy,
                                                 ValueRange{heapCopy, dynSize});
+      };
+
+      if (isa<IntegerType, FloatType>(eleTy)) {
+        std::size_t byteWidth = (eleTy.getIntOrFloatBitWidth() + 7) / 8;
+        result = createVectorInit(
+            builder.create<arith::ConstantIntOp>(loc, byteWidth, 64));
+      } else if (auto complexTy = dyn_cast<ComplexType>(eleTy)) {
+        eleTy = complexTy.getElementType();
+        if (!isa<IntegerType, FloatType>(eleTy)) {
+          TODO_x(toLocation(x), x, mangler, "unhandled complex element type");
+          return false;
+        }
+        std::size_t byteWidth = (eleTy.getIntOrFloatBitWidth() + 3) / 4;
```

Comment:
Q: Can you help me understand ``` ... + 3 ) / 4```? 
(Don't we need twice the size to account for real and imaginary parts)

Reply:
((x + 7) / 8) * 2 = (x + 3) / 4 in 2's complement.